[CS2113-W12-1] PlaNUS

ZhouJunmin opened the pull request:
PlaNUS serves as a centralized platform for our target user to manage their past, current, and future courses based on their grades and interests. The app is able to keep a record of the user’s existing academic record and give the user relevant course information and future course recommendations.

iscyng committed:
Updated Isaac-AboutUs

ZhouJunmin committed:
Update Aboutus.md

iscyng committed:
Merge pull request #2 from iscyng/Isaac-AboutUs

Updated Isaac-AboutUs

Hws2209 committed:
Update AboutUs.md

ZhouJunmin committed:
Merge branch 'Junmin-aboutus'

# Conflicts:
#	docs/AboutUs.md

Hws2209 committed:
Merge branch 'master' into HoWanSing-AboutUs

# Conflicts:
#	docs/AboutUs.md

Hws2209 committed:
Merge pull request #4 from Hws2209/HoWanSing-AboutUs

Update AboutUs.md

Hws2209 committed:
Update AboutUs

Hws2209 committed:
Merge pull request #5 from Hws2209/HoWanSing-AboutUs

Update AboutUs

ZhouJunmin committed:
Merge remote-tracking branch 'TeamRepo/master'

# Conflicts:
#	docs/AboutUs.md

ZhangWenyue3325 committed:
wenyue, about us

ZhouJunmin committed:
update aboutus.md again

ZhangWenyue3325 committed:
Wenyue

ZhangWenyue3325 committed:
Merge pull request #7 from ZhangWenyue3325/wenyue-aboutus

Wenyue aboutus

cirelesna committed:
Updated AboutUS.md

ZhouJunmin committed:
Merge remote-tracking branch 'TeamRepo/master' into AboutUsJunmin

# Conflicts:
#	docs/AboutUs.md

ZhouJunmin committed:
Merge pull request #6 from ZhouJunmin/AboutUsJunmin

About us junmin

cirelesna committed:
Merge branch 'Ansel-AboutUs'

# Conflicts:
#	docs/AboutUs.md

Hws2209 committed:
Add Course.java

Hws2209 committed:
Add Timetable.java

Hws2209 committed:
Add methods and Javadoc comments to Course class

Hws2209 committed:
Add methods and Javadoc comments to Timetable class

Hws2209 committed:
Move static attributes above instance-level attributes

Hws2209 committed:
Merge pull request #17 from Hws2209/master

Course and Timetable classes for v1.0

Hws2209 committed:
Add CourseTest.java

Hws2209 committed:
Add TimetableTest.java

Hws2209 committed:
Merge pull request #18 from Hws2209/master

Add JUnit tests for Course and Timetable classes

ZhouJunmin committed:
Create Storage Class

Storage enables PlaNUS to read from files and write to user data file.

ZhouJunmin committed:
Add header comments to Storage Class.

ZhouJunmin committed:
Storage Class debug

Change the order of "final" and "static"; declare variables separately.

ZhouJunmin committed:
Merge pull request #20 from ZhouJunmin/CreatingStorage

Create Storage Class

iscyng committed:
Added Ui class and added usage of ui

cirelesna committed:
Added PlaNus.java

cirelesna committed:
Merge pull request #22 from AY2324S2-CS2113-W12-1/Branch-MainClass

Added PlaNus.java

cirelesna committed:
Added Parser.java

cirelesna committed:
Added header comments to PlaNus.java and Parser.java

cirelesna committed:
Merge pull request #23 from AY2324S2-CS2113-W12-1/Branch-Parser

Branch parser

iscyng committed:
Added Ui class and added usage of ui

iscyng committed:
Added Ui class and added usage of ui

cirelesna committed:
Edited main to get application running

cirelesna committed:
Merge pull request #28 from AY2324S2-CS2113-W12-1/Branch-Parser

Edited main to get application running

cirelesna committed:
Fixed bug to get view function working

ZhangWenyue3325 committed:
Add in task related functins and JUnit test

ZhangWenyue3325 committed:
Merge pull request #29 from ZhangWenyue3325/features_related_to_grades

Add in task related functins and JUnit test

ZhouJunmin committed:
V1.0 Debug

ZhouJunmin committed:
v1.0 release

ZhouJunmin committed:
Merge pull request #30 from ZhouJunmin/branch_v1.0_debug

Branch v1.0 debug and release

cirelesna committed:
Enabled Assertions

cirelesna committed:
debug checkstyle error

Hws2209 committed:
Solve Checkstyle violations

Hws2209 committed:
Solve Checkstyle violations

Hws2209 committed:
Update EXPECTED.TXT and input.txt

Hws2209 committed:
Update build.gradle

Hws2209 committed:
Update EXPECTED.TXT and input.txt

Hws2209 committed:
Update EXPECTED.TXT

Hws2209 committed:
Update logo of PlaNUS to remove unnecessary spaces

Hws2209 committed:
Merge pull request #32 from Hws2209/master

Solve Java CI Checks failing

Hws2209 committed:
Add some assertions to Course and Timetable classes

Hws2209 committed:
Merge pull request #33 from Hws2209/v1.0-Assertions

Add some assertions to Course and Timetable classes

Hws2209 committed:
Add logger to PlaNus.java

Hws2209 committed:
Add usage of logging in Timetable class

Hws2209 committed:
List imported classes of logging explicitly

Hws2209 committed:
Update TimetableTest to match updated exception and logging message

Hws2209 committed:
Merge pull request #34 from Hws2209/v1.0-Logging

Set up logging

cirelesna committed:
Init method fix

cirelesna committed:
Merge pull request #36 from AY2324S2-CS2113-W12-1/Branch-FixInit

Init method fix

Hws2209 committed:
Change code to enable data files to be included in jar

Hws2209 committed:
Merge pull request #37 from Hws2209/IncludeDataFilesInJar

Change code to enable data files to be included in jar

Hws2209 committed:
refactor: Extract GradeChecker and PlanGetter from Timetable

Hws2209 committed:
Merge pull request #38 from Hws2209/master

refactor: Extract GradeChecker and PlanGetter from Timetable

Hws2209 committed:
Update v1.0 user guide

Hws2209 committed:
Merge pull request #39 from Hws2209/master

Update v1.0 user guide

ZhangWenyue3325 committed:
Add Assertions for grade-related methods.

ZhangWenyue3325 committed:
Merge pull request #40 from ZhangWenyue3325/master

Add Assertions for grade-related methods.

iscyng committed:
Modified parser such that it will accept grades in BOTH small and caps letter grades

iscyng committed:
Merge pull request #42 from AY2324S2-CS2113-W12-1/Isaac-ParserUpdate

Modified parser such that it will accept grades in BOTH small and cap…

ZhouJunmin committed:
Grade Checker Debug, Storage Assertions and Logging

ZhouJunmin committed:
Change Storage Class to meet Variable Declaration Order

ZhouJunmin committed:
Merge pull request #43 from ZhouJunmin/Branch_Grade_Checker_Debug

Grade Checker Debug, Storage Assertions and Logging

cirelesna committed:
JUnit tests for Parser Class

cirelesna committed:
Fix checkstyle error for import*

cirelesna committed:
Merge pull request #47 from AY2324S2-CS2113-W12-1/Branch-JUnit

JUnit tests for Parser Class

cirelesna committed:
Fix checkstyle error for import*

cirelesna committed:
Revert "Fix checkstyle error for import*"

This reverts commit 94ecbf9f81e13247b6762f97c39e7e3de4a15069.

cirelesna committed:
Add Logging and Assertion for Parser.java

cirelesna committed:
Merge pull request #50 from AY2324S2-CS2113-W12-1/Branch-ParserLoggingAndAssertions

Branch parser logging and assertions

Hws2209 committed:
First draft of Developer Guide

Hws2209 committed:
Merge pull request #52 from Hws2209/master

First draft of Developer Guide

ZhangWenyue3325 committed:
Allocate some code to the right author

ZhangWenyue3325 committed:
Merge pull request #53 from ZhangWenyue3325/Branch_V1.0_Code_Contribution

Allocate some code to the right author

ZhouJunmin committed:
Update Draft DG for v1.0 Storage Class

ZhangWenyue3325 committed:
Add logging for grades related methods, fixed the problem related to the maximum number of years and terms when checking grades

ZhouJunmin committed:
Merge pull request #54 from ZhouJunmin/Branch_DG_Storage_Update_1

Update Draft DG for v1.0 Storage Class

ZhangWenyue3325 committed:
Merge pull request #55 from ZhangWenyue3325/master

Add logging for grades related methods, fixed the problem related to …

ZhangWenyue3325 committed:
Add Developer Diagram for the grades related features, add documentation for grades related methods

ZhangWenyue3325 committed:
Merge pull request #56 from ZhangWenyue3325/Branch-DG

Add Developer Diagram for the grades related features, add documentat…

Hws2209 committed:
Update Developer Guide

Hws2209 committed:
Merge branch 'master' into master

Hws2209 committed:
Merge pull request #57 from Hws2209/master

Update Developer Guide

iscyng committed:
Modified Parser such that user can input specific courses with the specific MCs (to add their UEs or any mods that they want to take)

iscyng committed:
Modified Parser such that user can input specific courses with the specific MCs (to add their UEs or any mods that they want to take)

iscyng committed:
Merge pull request #58 from AY2324S2-CS2113-W12-1/Isaac-MCs

Added feature to allow user to specify MCs

Hws2209 committed:
Add function to display recommended schedules of different majors without init

Hws2209 committed:
Merge pull request #59 from Hws2209/master

Add function to display recommended schedules of different majors wit…

Hws2209 committed:
Suppress logging output to console

Hws2209 committed:
Merge pull request #60 from Hws2209/master

Suppress logging output to console

ZhangWenyue3325 committed:
Update JUnit Tests

ZhangWenyue3325 committed:
Merge pull request #61 from ZhangWenyue3325/Branch-DG

Update JUnit Tests

Hws2209 committed:
Update user guide for 'display' function

Hws2209 committed:
Merge pull request #62 from Hws2209/master

Update user guide for 'display' function

iscyng committed:
Modified addgrade to accept S/U/CS/CU

Hws2209 committed:
Allocate some code to author

Hws2209 committed:
Merge pull request #64 from Hws2209/master

Allocate some code to author

ZhangWenyue3325 committed:
Update Developer Guide

ZhangWenyue3325 committed:
fixed some bugs

ZhangWenyue3325 committed:
fixed test cases

ZhangWenyue3325 committed:
Merge pull request #65 from ZhangWenyue3325/Branch-SequenceDiagram

Update Developer Guide

iscyng committed:
Modified GradeChecker such that it will calculate GPA correctly with CSCUSU grades

iscyng committed:
Modified GradeChecker such that it will calculate GPA correctly with CSCUSU grades

iscyng committed:
Merge pull request #66 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Isaac sucscu

iscyng committed:
Modified UserGuide for the updated user inputs.

iscyng committed:
Merge pull request #67 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Modified UserGuide for the updated user inputs.

cirelesna committed:
Add functionality to load different user timetable

cirelesna committed:
Add functionality for user command to change between timetables

cirelesna committed:
Edit EXPECTED.TXT to match output

cirelesna committed:
Merge pull request #68 from AY2324S2-CS2113-W12-1/Branch-MultipleTimetables

Branch multiple timetables

iscyng committed:
Modified UserGuide for the updated user inputs.

Hws2209 committed:
Merge pull request #69 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Modified printHelp to accomadate for new user input changes

Hws2209 committed:
Update output of "help" command

Hws2209 committed:
Merge pull request #70 from Hws2209/master

Update output of "help" command

Hws2209 committed:
Update User Guide

Hws2209 committed:
Merge pull request #71 from Hws2209/master

Update User Guide

Hws2209 committed:
Update Storage.java

Hws2209 committed:
Merge pull request #72 from Hws2209/master

Update Storage.java

Hws2209 committed:
Add assertions to Course.java

Hws2209 committed:
Add assertions to PlanGetter.java

Hws2209 committed:
Update TimetableTest

Hws2209 committed:
Add PlanGetterTest.java

Hws2209 committed:
Merge pull request #75 from Hws2209/master

Add assertions and JUnit tests

Hws2209 committed:
Add more logging

Hws2209 committed:
Merge pull request #76 from Hws2209/master

Add more logging

cirelesna committed:
Add Parser component to DG

cirelesna committed:
Edited printHelp() to include change timetable

iscyng committed:
Merge pull request #77 from cirelesna/master

Edited printHelp() to include change timetable

iscyng committed:
Added Ui implementation and class diagram in DG

iscyng committed:
Edited changes for Ui Class Diagram

cirelesna committed:
Merge pull request #78 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Added Ui implementation and class diagram in DG

iscyng committed:
Added context to DG to further explain the design choice and integration with other components

cirelesna committed:
Merge pull request #79 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Added context to DG to further explain the design choice and integrat…

iscyng committed:
Added and edited assertions to GradeChecker

iscyng committed:
Added and edited assertions to GradeChecker

iscyng committed:
modified added assertions to planus

cirelesna committed:
Merge pull request #81 from AY2324S2-CS2113-W12-1/Isaac-SUCSCU

Add assertions to GradeChecker

ZhouJunmin committed:
Fixed bugs related to grade checker with SU modules

ZhouJunmin committed:
Merge pull request #83 from ZhouJunmin/Debug_GradeChecker_v2.0

Fixed bugs related to grade checker with SU modules

ZhouJunmin committed:
Add automatic fill of course name given course code when adding courses

ZhouJunmin committed:
Delete unused import scanner in parser class

ZhouJunmin committed:
Merge pull request #84 from ZhouJunmin/Branch_autoAddCourseName_v2.0

Add automatic fill of course name given course code when adding courses

ZhouJunmin committed:
Modified course adding function to prevent adding duplicated modules into the timetable.

ZhouJunmin committed:
Merge pull request #85 from ZhouJunmin/Branch_optimize_addcourse_v2.0

Modified course adding function

ZhouJunmin committed:
add move course function

ZhouJunmin committed:
Merge pull request #87 from ZhouJunmin/Branch_move_course

add move course function

ZhangWenyue3325 committed:
Refine Chatbot responses and make PlaNUS more fool-proof

ZhangWenyue3325 committed:
Update uitest

ZhangWenyue3325 committed:
Merge pull request #88 from ZhangWenyue3325/Branch_UI_v2.0

Refine Chatbot responses and make PlaNUS more fool-proof

ZhouJunmin committed:
Add in unit tests and header comments for Storage Class v2.0

ZhouJunmin committed:
Change imports of StorageTest class to satisfy style check

ZhouJunmin committed:
Merge pull request #89 from ZhouJunmin/Branch_v2.0_Storage_TestAndDocumentation

Add in unit tests and header comments for Storage Class v2.0

iscyng committed:
Added GradeCheckerTest to ensure working functionality of GPA calculation

cirelesna committed:
Debug checkstyle error

cirelesna committed:
Edit failing GradeCheckerTest

cirelesna committed:
Merge pull request #90 from cirelesna/master

Debug checkstyle error

ZhouJunmin committed:
Update move course feature to User Guide.

ZhouJunmin committed:
Merge pull request #91 from ZhouJunmin/Branch_update_UG_moveCourse

Update move course feature to User Guide.

ZhouJunmin committed:
Update Storage Class to include data files in Jar file build.

ZhouJunmin committed:
Merge pull request #92 from ZhouJunmin/Branch_v2.0_Release

Update Storage Class to include data files in Jar file build.

ZhangWenyue3325 committed:
Update User Guide

ZhangWenyue3325 committed:
Merge pull request #93 from ZhangWenyue3325/master

Update User Guide

iscyng committed:
Modified Parser to standardise the spacing between commands. Updated DG and UG

iscyng committed:
Modified indentation

iscyng committed:
Checkstyle error

iscyng committed:
Merge pull request #95 from iscyng/master

Modified Parser to standardise the spacing between commands. Updated …

annoy-o-mus commented:
Does the access modifiers of  the class diagrams used in this DG consistent with that taught in the course (`+` instead of a solid green circle, the green circle at the top etc)

annoy-o-mus commented:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/55198236/a130a460-e2fa-41b9-92eb-8f01de8709c1)

If there is only a single option (`if` without `else`) would `opt` be a better alternative to `alt`?

e0958902 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Parser component
+The Parser is essential to interpreting user commands and executing appropriate actions on the timetable. 
```
This developer guide looks neat.

e0958902 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Below is the class diagram:
+
+![Parser_ClassDiagram](uml/Parser_ClassDiagram.png)
```
Id suggest that the different classes in the class diagram do not have this symbol 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/187285de-d0ef-41a4-b6a2-b04772d900ed)
Also, the public are to be used with "+" and private is with "-" 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/45c881b1-6db7-464d-a9ae-76142febeedd)

annoy-o-mus commented:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/55198236/efee13d8-2da6-406b-b6da-0599b29c895e)

For the part highlighted in yellow, it should be an instance (an object) or the class itself, rather than what seems like an attribute of the class. Can perhaps consider invoking a self-call instead

e0958902 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Below is a sequence diagram that shows how Storage takes part in the programme.
+
+![Storage_SequenceDiagram](uml/Storage_SequenceDiagram-0.png)
```
Adding a << class >> in the sequence diagram indicates that it is an class. Alternatively, you could also use ":Storage". I have asked around and both is fine. Good job!
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/0aaf22b2-77bc-4ee9-9f61-85dad2227f25)

annoy-o-mus commented:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/55198236/8042943f-190e-4a6c-9ce6-428f02fb4c48)

The class here by itself seems unnecessary as it does not show how it relates to anything else

e0958902 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddGrade_SequenceDiagram](uml/AddGrade_SequenceDiagram-0.png)
```
The Boolean to check may seem too lengthy. Perhaps, something like matchCourseCode would be easier to read. Then the Boolean can be further explained in a separate comment.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/156929749/1376c8a4-d36c-4ff6-b720-c48ba2ca64d1)

e0958902 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Below is the sequence diagram to illustrate the implementation:
+
+![RemoveGrade_SequenceDiagram](uml/RemoveGrade_SequenceDiagram-0.png)
```
Same as above mentioned.

e0958902 reviewed (commented):
These are the issues or comments I may have. Hope all these would help.
You have helped me learn and apply these common issues for my future developer guides.

soongensayo commented:
There's clear delineation of the UI component's key methods and their purposes, which helps in understanding how user interactions are handled!
<img width="426" alt="Screenshot 2024-04-04 at 11 12 29 AM" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/110648719/409779e9-0845-4d78-a83c-e09e0182b87d">

soongensayo commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+*   **Storage Component**: Notifications about file operations (e.g., file creation success or failure) are relayed to 
+the user via the Ui.
+
+![Ui_ClassDiagram](uml/UiClassDiagram.png)
```
There's clear delineation of the UI component's key methods and their purposes, which helps in understanding how user interactions are handled.

1simjustin replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/13176823/9ef6f5af-b0c2-4cfe-9dec-e393e1f27b0c)
This diagram might possibly contain too much details which can be summarised to improve readability.

soongensayo commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+The first only takes in Timetable `timetable`, which records the current plan of the user.
+The second takes in an additional integer `year`, while the third takes in two additional integers `year` and `term`.
+
+### Ui Component
```
Consider providing code examples or scenarios for when each UI method might be used to offer practical context :)

soongensayo commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+<br>
+
+There are 3 overloaded variations of the `getPlan()` method in the `PlanGetter` class.
```
The overloading of the getPlan() method shows consideration for different user needs and scenarios, reflecting well-thought-out software design principles.

1simjustin replied:
Would it have been better to introduce these overloaded methods in the above diagram as well? Otherwise, since it is referencing "the above diagram", it may be better to position this section right under the class diagram for ease of reference as well.

soongensayo commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+<br>
+
+There are 3 overloaded variations of the `getPlan()` method in the `PlanGetter` class.
+The first is as shown in the class diagram above. 
```
The complexity introduced by having three getPlan() variations could be mitigated by explaining why it's necessary and how each contributes to the system's functionality.

soongensayo commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+If target file is not found, the above two methods will be able to create the missing file.
+
+Storage also contains a useful parser method `parseCourse(String)` that can parse a line of csv file into a Course object.
```
Details on how parseCourse(String) manages diverse CSV formats would ensure robust parsing and contribute to overall system resilience against data format inconsistencies.

1simjustin commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
 
 ## Design & implementation
 
```
Would it be better to include a high level overview?

1simjustin commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+act upon.
 
+Parser interacts with the following classes.
+- 'Course': Creates Courses based on user input
```
Was there an issue with applying markdown in this section?

1simjustin commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Storage has two main file access APIs:
+- `writeToFile(Timetable)`:
+  - Take in a timetable containing courses, then write courses to the user data file at __./data/myTimetable.csv__.
```
Use of markdown here to bold the file directory is slightly jarring, might be better to use code section or code block markdowns instead.

1simjustin reviewed (commented):
Overall the DG looks neat! The use of diagrams is apt. Good job!

Hws2209 committed:
Update data files

Hws2209 committed:
Merge pull request #97 from Hws2209/master

Update data files

Hws2209 committed:
Solve some bugs

Hws2209 committed:
Merge pull request #98 from Hws2209/master

Solve some bugs

owx0130 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+
+The grades keyed in by the user are **String** type, e.g.`"A"`.
+In the `Grade` class, the `setNumberGrade()` method converts the grade into **double** type, which is more appropriate for the calculation of GPA and total grades.
```
Might be more readable if the `double` and `String` word here is enclosed within the code markdown instead of bolding it

owx0130 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
 
 ## Design & implementation
```
Maybe you could remove the "C" symbol in all your class diagrams to adhere to what is taught in lecture

owx0130 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+
+Below is the sequence diagram to illustrate the implementation:
+
+![AddCourse_SequenceDiagram](uml/AddCourse_SequenceDiagram.png)
```
Your sequence diagrams should have dashed return arrows even if it returns `void` just to show that the function call is completed

owx0130 commented on docs/DeveloperGuide.md:
```diff
@@ -6,24 +6,177 @@
+Below is the sequence diagram to illustrate the implementation:
+
+![AddCourse_SequenceDiagram](uml/AddCourse_SequenceDiagram.png)
+
```
Also, objects in the sequence diagram should only be labelled at the top of its lifeline, there should not be any labelling below the lifeline of the object

iscyng committed:
Fix bug that allowed users to view years outside the range given

Hws2209 committed:
Debug some code according to PE-D comments

Hws2209 committed:
Update UG and DG according to comments given

Hws2209 committed:
Merge pull request #133 from Hws2209/master

Solve some issues

Hws2209 committed:
Update data files to remove UE/GE

Hws2209 committed:
Merge pull request #134 from Hws2209/master

Update data files to remove UE/GE

iscyng committed:
Fix bug such that app blocks viewing by term only

iscyng committed:
Fix bug such that app blocks checking by term only

iscyng committed:
Fix bug that causes change grade sets grade to null if invalid grade is input

Hws2209 committed:
Fix buggy data

Hws2209 committed:
Merge pull request #135 from Hws2209/master

Fix buggy data

Hws2209 committed:
Add hws2209.md, Update README.md and AboutUs.md

Hws2209 committed:
Merge pull request #136 from Hws2209/master

Add hws2209.md, Update README.md and AboutUs.md

iscyng committed:
Fconflict

iscyng committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp:
  Add hws2209.md, Update README.md and AboutUs.md
  Fix buggy data

Hws2209 committed:
Ensure that optional parameters format (e.g. "y/") are verified

Hws2209 committed:
Merge pull request #137 from Hws2209/master

Ensure that optional parameters format (e.g. "y/") are verified

Hws2209 committed:
Update Parser.java

Hws2209 committed:
Merge pull request #138 from Hws2209/master

Update Parser.java

Hws2209 committed:
Update DG

Hws2209 committed:
Merge pull request #139 from Hws2209/master

Update DG

Hws2209 committed:
Solve buggy data related to ChBE

Hws2209 committed:
Merge pull request #140 from Hws2209/master

Solve buggy data related to ChBE

iscyng committed:
PPP-Draft and updated Develoepr Guide with Architecture diagram

Hws2209 committed:
Add Wan Sing's Project Portfolio Page

Hws2209 committed:
Merge pull request #141 from Hws2209/master

Add Wan Sing's Project Portfolio Page

cirelesna committed:
Edited default checkGrade to display 4 years unless courses available in year 5 or 6

cirelesna committed:
Merge pull request #142 from cirelesna/master

Edited default checkGrade to display 4 years

cirelesna committed:
Edited format of Parser_ClassDiagram.puml

cirelesna committed:
Merge pull request #143 from cirelesna/master

Edited format of Parser_ClassDiagram.puml

ZhangWenyue3325 committed:
Set GPA to 2 decimal place

ZhangWenyue3325 committed:
Change the desired output to 2dp

ZhangWenyue3325 committed:
solve merging problems

ZhangWenyue3325 committed:
Merge pull request #144 from ZhangWenyue3325/master

Set GPA to 2 decimal place

ZhangWenyue3325 committed:
Add PPP draft 1 for wenyue

ZhangWenyue3325 committed:
Merge pull request #145 from ZhangWenyue3325/master

Add PPP draft 1 for wenyue

cirelesna committed:
Added sequence diagram for Parser for init command

cirelesna committed:
Merge pull request #146 from cirelesna/master

Added sequence diagram for Parser for init command

cirelesna committed:
Added PPP draft for cirelesna

cirelesna committed:
Merge pull request #147 from cirelesna/master

Added PPP draft for cirelesna

ZhouJunmin committed:
Fixed bugs related to misprinting error messages of integer overflow.

ZhouJunmin committed:
Merge pull request #148 from ZhouJunmin/V2.1_Debug_Integer_Overflow

Fixed bugs related to misprinting error messages of integer overflow.

ZhouJunmin committed:
Fixed bugs related to commas

The presence of commas in the user input course code and course name could lead to data corruption in csv files.

ZhouJunmin committed:
Merge pull request #149 from ZhouJunmin/V2.1_Debug_Comma

Fixed bugs related to commas

ZhouJunmin committed:
Update DG

Modify Storage component of DG based on feedbacks.

ZhouJunmin committed:
Merge pull request #150 from ZhouJunmin/V2.1_Modify_DG

Update DG

ZhouJunmin committed:
Update PPP Draft

ZhouJunmin committed:
Merge pull request #151 from ZhouJunmin/V2.1_PPP_Draft

Update PPP Draft

ZhangWenyue3325 committed:
Edit the deeply nested GradeChecker

ZhangWenyue3325 committed:
Merge pull request #153 from ZhangWenyue3325/master

Edit the deeply nested GradeChecker

Hws2209 committed:
Update DG

Hws2209 committed:
Merge pull request #154 from Hws2209/master

Update DG

iscyng committed:
Updated UIClassDiagram

cirelesna committed:
Edited formatting for Grade class diagram

cirelesna committed:
Merge pull request #155 from cirelesna/master

Edited formatting for Grade class diagram

cirelesna committed:
Edited formatting for Grade class diagram

cirelesna committed:
Edited formatting for Grade class diagram

cirelesna committed:
Merge pull request #156 from cirelesna/master

Edit Class diagram

iscyng committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp:
  Edited formatting for Grade class diagram
  Edited formatting for Grade class diagram
  Edited formatting for Grade class diagram

ZhouJunmin committed:
Rename ZhouJunmin.md to zhoujunmin.md

iscyng committed:
Updated UG

ZhouJunmin committed:
Rename ZhangWenyue3325.md to zhangwenyue3325.md

iscyng committed:
Merge commit '412e8872382e66b450b13c589600226e9bbfff89'

* commit '412e8872382e66b450b13c589600226e9bbfff89':
  Rename ZhouJunmin.md to zhoujunmin.md

iscyng committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-W12-1/tp:
  Rename ZhangWenyue3325.md to zhangwenyue3325.md
